tools/base/checker.py

```diff
@@ -0,0 +1,276 @@
+            type=int,
+            default=5,
+            help="Number of errors to show in the summary, -1 shows all")
+        parser.add_argument(
+            "--summary-warnings",
+            type=int,
+            default=5,
+            help="Number of warnings to show in the summary, -1 shows all")
+        parser.add_argument(
+            "--check",
+            "-c",
+            choices=self.checks,
+            nargs="*",
+            help="Specify which checks to run, can be specified for multiple checks")
+        for check in self.checks:
+            parser.add_argument(
+                f"--config-{check}", default="", help=f"Custom configuration for the {check} check")
+        parser.add_argument(
+            "--path",
+            "-p",
+            default=None,
+            help=
+            "Path to the test root (usually Envoy source dir). If not specified the first path of paths is used"
+        )
+        parser.add_argument(
+            "--log-level",
+            "-l",
+            choices=["info", "warn", "debug", "error"],
+            default="info",
+            help="Log level to display")
+        parser.add_argument(
+            "paths",
+            nargs="*",
+            help=
+            "Paths to check. At least one path must be specified, or the `path` argument should be provided"
+        )
+        return parser
+
+    @cached_property
+    def path(self) -> str:
+        try:
+            path = self.args.path or self.args.paths[0]
+        except IndexError:
+            raise self.parser.error(
+                "Missing path: `path` must be set either as an arg or with --path")
+        if not os.path.isdir(path):
+            raise self.parser.error(
+                "Incorrect path: `path` must be a directory, either as first arg or with --path")
+        return path
+
+    @property
+    def paths(self) -> list:
+        return self.args.paths or [self.path]
+
+    @property
+    def recurse(self) -> bool:
+        return self.args.recurse
+
+    @property
+    def show_summary(self) -> bool:
+        return bool(self.args.summary or self.error_count or self.warning_count)
+
+    @property
+    def status(self) -> dict:
+        return dict(
+            success=self.success_count,
+            errors=self.error_count,
+            warnings=self.warning_count,
+            failed=self.failed,
+            warned=self.warned,
+            succeeded=self.succeeded)
+
+    @property
+    def succeeded(self) -> dict:
+        return dict((k, (len(v))) for k, v in self.success.items())
+
+    @property
+    def success_count(self) -> int:
+        return sum(len(e) for e in self.success.values())
+
+    @property
+    def warned(self) -> dict:
+        return dict((k, (len(v))) for k, v in self.warnings.items())
+
+    @property
+    def warning_count(self) -> int:
+        return sum(len(e) for e in self.warnings.values())
+
+    def bazel_query(self, query: str) -> list:
```

Comment:
not sure if bazel can be queried directly in python

also move this to a `BazelChecker` because its a dep not all checkers will want